src/main/groovy/net/wasdev/wlp/gradle/plugins/tasks/InstallLibertyTask.groovy

```diff
@@ -134,4 +135,38 @@ class InstallLibertyTask extends AbstractTask {
             }
         }
     }
+    
+    
+    protected boolean isInstalled() {
+        boolean isUpToDate = false;
+
+        File f = new File(project.buildDir, 'liberty-plugin-config.xml')
+        if (f == null || !f.exists()) {
+            return isUpToDate;
+        }
+
+        XmlParser pluginXmlParser = new XmlParser()
+        Node libertyPluginConfig = pluginXmlParser.parse(f)
+
+        if(project.configurations.libertyRuntime != null && !project.configurations.libertyRuntime.dependencies.isEmpty()) {
+            if (!libertyPluginConfig.getAt('assemblyArtifact').isEmpty()) {
+                project.configurations.libertyRuntime.dependencies.each { libertyArtifact ->
+                    if(libertyPluginConfig.getAt('assemblyArtifact').getAt('artifactId').text().equals(libertyArtifact.name)
+                        && libertyPluginConfig.getAt('assemblyArtifact').getAt('version').text().equals(libertyArtifact.version)
+                        && libertyPluginConfig.getAt('assemblyArtifact').getAt('groupId').text().equals(libertyArtifact.group)) {
+                            isUpToDate = true;
```

Comment:
So doesn't this mean that if any of the libertyArtifact is the same, it reports that it is up to date regardless of the rest of the libertyArtifacts? It seems like it should check all of them, and only if all of them are the same then it is up to date.

Reply:
There should only ever be one libertyArtifact specified, but if there is more than one, I believe we are only worried about the first one specified. But agree, logic should change to check only once instead